database/migrations/2024_08_01_201721_add_required_notes_setting.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('settings', function (Blueprint $table) {
+            $table->boolean('require_checkinout_notes')->nullable()->default(0);
```

Comment:
I think it'll be good to add this field to the [Setting's models `$casts`](https://github.com/snipe/snipe-it/blob/3167ee91d17e11b86fcad450319bc806e25b06f2/app/Models/Setting.php#L101) so we can use it in this PR and in the future as

```php
$settings->require_checkinout_notes
```

and not

```php
$settings->require_checkinout_notes == "1"
```

It makes the code easier to parse _and_ gets around type-juggling.

[docs](https://laravel.com/docs/10.x/eloquent-mutators#attribute-casting).